extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/MongoConnection.java

```diff
@@ -109,7 +109,7 @@ boolean reconnectIfNecessary() {
                         + willRetryMessage(), e);
                 return false;
             } catch (Exception e) {
-                throw new JetException("Cannot connect to MongoDB, seems to be non-transient error", e);
+                throw new JetException("Cannot connect to MongoDB, seems to be non-transient error: " + e.getMessage(), e);
```

Comment:
Is this change helpful? The message should be in the stacktrace.

Reply:
Error messages are unfortunatelly not visible in some cases. They are still on server side, but on client side I sometimes got only upper-error message (e.g. when testing on Viridian - yeah I could take a look at server logs and search the error, but it's faster and easier to have the root cause visible directly)

Reply:
Right, the client not having full exception

Could you comment on it here, thanks
https://hazelcast.atlassian.net/browse/HZ-2383

Reply:
Done